### PR TITLE
LoginView SwiftUI

### PR DESCRIPTION
**Pull request by barbarakos**
Created a login screen in SwiftUI .
Simulator Screen Shot - iPhone 14 Plus - 2022-12-08 at 10 31 56

### CODE REVIEW

File: QuizApp/Login/LoginView.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    @FocusState private var passwordIsFocused: Bool
+
+    var viewModel: LoginViewModel
+
+    var body: some View {
+        ZStack {
+            LinearGradient(gradient:
+                            Gradient(colors:
+                                        [Color(red: 0.453, green: 0.308, blue: 0.637),
+                                         Color(red: 0.154, green: 0.185, blue: 0.463)]),
```
Comment (marinzg) [inline QuizApp/Login/LoginView.swift:16]: Can this be a private computed property?

File: QuizApp/Login/LoginView.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    .focused($emailIsFocused)
+                    .foregroundColor(.white)
+                    .frame(width: 300, height: 50)
+                    .background(Color.white.opacity(0.3))
+                    .cornerRadius(15)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 15)
+                            .stroke(.white, lineWidth: emailIsFocused ? 1 : 0)
+                    )
+                    .padding(.top, 60)
```
Comment (marinzg) [inline QuizApp/Login/LoginView.swift]: Indent one tab.

File: QuizApp/Login/LoginView.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    var loginFieldsValid: Bool {
+        return !password.isEmpty && !username.isEmpty
+    }
+
+    var buttonColor: Color {
+        return loginFieldsValid ? .white.opacity(1) : .white.opacity(0.6)
+    }
```
Comment (marinzg) [inline QuizApp/Login/LoginView.swift]: You can move this logic to `ViewModel` and make them `@Published` and change view model to `@ObservedObject`.
That way, you can use binding from view to view model to calculate is the field valid, and the same binding in button color so you don't have to have these variables.
```swift
// LoginView

...
TextField("Email", text: $viewModel.username)
...

Button("Login") {
        viewModel.login(username: username, password: password)
    }
    .background($viewModel.loginFieldsValid ? .white.opacity(1) : .white.opacity(0.6))
    .disabled(!$viewModel.loginFieldsValid)

// LoginViewModel

@Published var username: String = ""
@Published var loginFieldsValid: Bool = false

...
```

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Login/LoginView.swift (hunk 1 above)
Comment (barbarakos) [reply QuizApp/Login/LoginView.swift:16]: It is changed in the next PR (#32 )

**Review by marinzg — APPROVED** [review]

**Pull request merged**